pkgs/servers/nosql/mongodb/v4_0.nix

```diff
@@ -11,7 +11,15 @@ let
 in buildMongoDB {
   version = "4.0.27";
   sha256 = "sha256-ct33mnK4pszhYM4Is7j0GZQRyi8i8Qmy0wcklyq5LjM=";
-  patches =
-    [ ./forget-build-dependencies.patch ./mozjs-45_fix-3-byte-opcode.patch ]
+  patches = [
+    ./forget-build-dependencies.patch
+    ./mozjs-45_fix-3-byte-opcode.patch
+    ./mongodb-4.0-glibc-2.34.patch # https://github.com/NixOS/nixpkgs/issues/171928
+   (fetchpatch {
+      name = "mongodb-4.4.1-gcc11.patch";
```

Comment:
Patch doesn't apply on 4.0, is there a reason we want to keep it around? My proposal is to just get rid of version 4.0.

Reply:
Without the patch, it does not build, also I think the patch is more related to GCC11 than MongoDB version itself.

And yes, MongoDB 4.0 is already EOLed, but I think deprecating it now on Nixpkgs is not a good idea (it deprecated only last month).

Reply:
A reason to keep it around for the next NixOS release was discussed here: https://github.com/NixOS/nixpkgs/issues/155121#issuecomment-1120304376

Reply:
@andersk https://github.com/NixOS/nixpkgs/issues/155121#issuecomment-1120304376 is a good reason to keep it around.

@bryanasdev000 Sorry my mistake I missread the error.  It only fails on darwin and it is the `asio-no-experimental-string-view.patch` patch (darwin only) that fails to apply.
```
       last 10 log lines:
       > applying patch /nix/store/k0xlii3xyxlfparmx0w9ylmc75swgfbc-mongodb-4.0-glibc-2.34.patch
       > patching file src/mongo/stdx/thread.h
       > applying patch /nix/store/pmzplzw7nd78jrc3wf9y2jdarww58q6y-mongodb-4.4.1-gcc11.patch
       > patching file src/mongo/db/exec/plan_stats.h
       > Hunk #1 succeeded at 34 with fuzz 2 (offset 1 line).
       > applying patch /nix/store/f57czlnkrc71v6dh5k76msr3nc6jw12s-asio-no-experimental-string-view.patch
       > patching file src/third_party/asio-master/asio/include/asio/detail/config.hpp
       > Hunk #1 FAILED at 786.
       > Hunk #2 FAILED at 794.
       > 2 out of 2 hunks FAILED -- saving rejects to file src/third_party/asio-master/asio/include/asio/detail/config.hpp.rej

```